### ISSUE TITLE
Routing to Search/ crashes the application

### ISSUE THREAD

**Issue by liwzhang** [autoinvent/conveyor#148]
Trying to go to Search/ will cause the application to crash. This happens when clicking on the search buton with an empty query, as it's a `Link` component with a `to` prop that routes directly to `Search/${queryText}`. Hitting "enter" with an empty query won't trigger the error because routing is done with `useHistory.push()` after a conditional to check if query is empty.

A possible temporary fix is to switch out the `Link` component of the search button to another component that allows for a callback function to check if query is empty, and then route with `useHistory.push()`. However a possible permanent fix for this would be to either redirect to a 404 page or to simply display the search page upon visiting Search/. 

Edit: Another related issue is that going directly to Search/book for example would show no search results.
![image](https://user-images.githubusercontent.com/22905812/124327480-d69bfd80-db3c-11eb-80bc-c259bb200767.png)

**Comment by DanielHall977** [comment]
Ok, 2 separate issues.
1. The crashing, this is due to the router getting confused with the empty case and not routing to the search page. It can be fixed by using `?query=` instead of the path like every other search engine does. Or alternatively, it can be worked around by adding
```
<Route exact path="/Search/" component={SearchPageContainer} />
```
to the main pages routing section.

2. Going straight to the search page through the url doesn't return any results because the 'search the backend' epics are not run when it the page is loaded.

**Comment by itechify** [comment]
Resolved #155
